Cython/Compiler/Symtab.py

```diff
@@ -2123,8 +2123,6 @@ def declare_var(self, name, type, pos,
         elif type.needs_refcounting:
             if not allow_refcounted:
                 error(pos, "C struct/union member cannot be reference-counted type '%s'" % type)
-        if visibility != 'private':
-            error(pos, "C struct/union member cannot be declared %s" % visibility)
```

Comment:
@matusvalo Could you explain/provide an example of why you had to remove this check?

Reply:
It was removed because now we allow non private visibility. But I agree that it should be adjusted. I sm preparing the change.

Reply:
>  Isn't it just that the member visibility must match the visibility of the struct/union itself?

This sounds reasonable. I have implemented it. I am not 100% sure whether it is correct to store struct/union visibility in the Scope object but this seemed to be most simple and safe solution. Let me know.

Reply:
@scoder I've been looking with more detail into this.

If we follow your suggestion of using `body_ctx = Ctx(visibility=ctx.visibility)` (which I totally agree), then the check for visibility seems rather redundant: the Cython syntax simply does not allow a struct member to be declared with any visibility.

I believe the true branch of the original check never ever triggered. We have no test checking for the error. Moreover, I tried to come up with a test that would trigger the error, but I failed. Can you produce a test that would emit the error? Otherwise, maybe it is simpler to just remove the check as @matusvalo did previously?

Reply:
I don't know, is it possible to declare e.g. function pointers in struct fields as `api` or `public`?

Reply:
> is it possible to declare e.g. function pointers in struct fields as api or public?

No, it is not. The Cython syntax does not support that. If you try, you would get a parsing error.

Reply:
> If you try, you would get a parsing error.

Ok, then there is really not much use in having the check in place.

There is another method in the scope class, though: `declare_cfunction()`. It can receive a visibility and passes it on into `declare_var()`. Does anyone know if that is for function pointers or something else? C++ struct methods, maybe?

Reply:
> Ok, then there is really not much use in having the check in place.

Hm the PR was merged with check in place. Can I create separate PR removing it?

Reply:
I removed the check in https://github.com/cython/cython/commit/fd3b9bbaf3a8e4c83dd18a4ce099607b38d66933

Reply:
Thank you @scoder !